src/contractor/contractor.cpp

```diff
@@ -81,6 +81,38 @@ inline EdgeWeight distanceAndSpeedToWeight(double distance_in_meters, double spe
     return std::max<EdgeWeight>(1, static_cast<EdgeWeight>(std::round(duration * 10)));
 }
 
+// Returns updated edge weight
+template <class IterType>
+EdgeWeight getNewWeight(IterType new_speed_iter,
+                        const double &segment_length,
+                        const std::vector<std::string> &segment_speed_filenames,
+                        EdgeWeight oldWeight,
+                        const double &verify_weights)
+{
+    auto new_segment_weight =
+        (new_speed_iter->speed_source.speed > 0)
+            ? distanceAndSpeedToWeight(segment_length, new_speed_iter->speed_source.speed)
+            : INVALID_EDGE_WEIGHT;
+    // the check here is enabled by the `--verify-weights` flag
+    // it logs a warning if the new weight exceeds a heuristic of what a reasonable weight update is
+    if (verify_weights > 0)
+    {
+        if ((new_segment_weight < oldWeight) && ((oldWeight / new_segment_weight) > verify_weights))
```

Comment:
If you read like 7 lines up, it's the check that happens while `new_segment_weight` is defined, https://github.com/Project-OSRM/osrm-backend/pull/2906/files/566c96b258efc369132bc10e09df4c2d2fd136e2#diff-5d25f965a92c5af3a4dfd03cf4f7616aR93